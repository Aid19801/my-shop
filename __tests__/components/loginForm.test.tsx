import { fireEvent, render, screen } from "@testing-library/react";
import LoginForm from "../../components/login-form";

const MOCK_USERNAME = "foo-name";
const MOCK_PASSWORD = "foo-pw12";

describe("LoginForm", () => {
  it("should render the component without exploding", () => {
    render(<LoginForm onSubmit={() => null} errorMsg="foo" />);
    const form = screen.getByTestId("login-form");
    expect(form).toBeInTheDocument();
  });
  it("should match snapshot", () => {
    const { container } = render(
      <LoginForm onSubmit={() => null} errorMsg="foo" />
    );
    expect(container).toMatchSnapshot();
  });
  it("should capture the user's username as expected", () => {
    render(<LoginForm onSubmit={() => null} errorMsg="foo-err" />);
    const input = screen.getByPlaceholderText("username");
    fireEvent.change(input, { target: { value: MOCK_USERNAME } });
    // @ts-ignore
    expect(input.value).toBe(MOCK_USERNAME);
  });
  it("should capture the user's password as expected", () => {
    render(<LoginForm onSubmit={() => null} errorMsg="foo-err" />);
    const input = screen.getByPlaceholderText("password");
    fireEvent.change(input, { target: { value: MOCK_PASSWORD } });
    // @ts-ignore
    expect(input.value).toBe(MOCK_PASSWORD);
  });
});
