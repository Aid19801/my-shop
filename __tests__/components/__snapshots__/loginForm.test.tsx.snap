// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match snapshot 1`] = `
<div>
  <form
    data-testid="login-form"
  >
    <input
      name="username"
      placeholder="username"
    />
    <input
      name="password"
      placeholder="password"
    />
    <p>
      foo
    </p>
    <input
      type="submit"
    />
  </form>
  <style>
    
        form {
          display: flex;
          flex-direction: column;
        }
        input {
          height: 30px;
          margin-bottom: 30px;
        }
        .form__validationError {
          color: red;
          margin-top: -12px;
        }
      
  </style>
</div>
`;
